### PR TITLE
refactor(manager): replace share s3 flow with self-contained instructions

### PR DESCRIPTION
**Pull request by daniellacosse**
update the email invite text in favor of the invite flow, which should address the following ROS issues:

- https://git.radicallyopensecurity.com/ros/google-outline/-/issues/16
- https://git.radicallyopensecurity.com/ros/google-outline/-/issues/17

### CODE REVIEW

File: src/server_manager/messages/master_messages.json [hunk 1]
```diff
@@ -1243,9 +1243,69 @@
       }
     }
   },
-  "share_invite": {
-    "message": "You\u2019re invited to connect to my Outline server. Use it to access the open internet, no matter where you are. Follow the instructions on your invitation link below to download the Outline App and get connected.",
-    "description": "This string appears as a paragraph within a dialog that provides an invitation to share access to a server. Belongs to the instructions for the user with whom access is being shared."
+  "share_invite_intro": {
+    "message": "You’re invited to Outline! Follow these instructions to use my server to access the open internet on any device, no matter where you are. No personal information will be collected.",
+    "description": "TODO"
```
Comment (daniellacosse) [inline src/server_manager/messages/master_messages.json:1248]: Still need to fill these in. Requesting a review in advance due to time constraints.

File: src/server_manager/web_app/ui_components/outline-share-dialog.ts [hunk 2]
```diff
@@ -112,24 +112,51 @@ Polymer({
         ></p>
       </div>
       <div contenteditable="" id="selectableText" style="-webkit-text-size-adjust: 100%;">
-        <p>[[localize('share-invite')]]</p>
-
-        <p><a href$="{{s3Url}}">{{s3Url}}</a></p>
-        <p>-----</p>
-        <p>[[localize('share-invite-trouble')]]</p>
+        <p>[[localize('share-invite-intro')]]</p>
+        <p>[[localize('share-invite-step-one')]]</p>
```
Comment (fortuna) [inline src/server_manager/web_app/ui_components/outline-share-dialog.ts:116]: Create a ordered list for the steps.

File: src/server_manager/web_app/app.ts [hunk 3]
```diff
@@ -290,11 +290,12 @@ export class App {
 
     appRoot.addEventListener('OpenShareDialogRequested', (event: CustomEvent) => {
       const accessKey = event.detail.accessKey;
-      this.appRoot.openShareDialog(accessKey, this.getS3InviteUrl(accessKey));
+      this.appRoot.openShareDialog(accessKey);
     });
 
     appRoot.addEventListener('OpenGetConnectedDialogRequested', (event: CustomEvent) => {
-      this.appRoot.openGetConnectedDialog(this.getS3InviteUrl(event.detail.accessKey, true));
```
Comment (fortuna) [inline src/server_manager/web_app/app.ts:297]: We shouldn't have the get connected button take to the share dialog. 

The change in behavior should be accompanied by the actual change of the manager key UI. It should delete events and image resources. It's independent of how the share ui looks like, so let's put it in a separate PR.

**Review by fortuna — CHANGES_REQUESTED** [review]

File: src/server_manager/messages/en.json [hunk 4]
```diff
@@ -263,12 +263,22 @@
...
   "setup-title": "Choose a cloud service to set up Outline.",
   "share-description": "Copy this invitation and send it from a communication tool you trust. {openLink}Need help?{closeLink}",
-  "share-invite": "You’re invited to connect to my Outline server. Use it to access the open internet, no matter where you are. Follow the instructions on your invitation link below to download the Outline App and get connected.",
   "share-invite-copied": "Copied to clipboard",
   "share-invite-copy": "Copy invitation",
-  "share-invite-copy-access-key": "Copy your access key:",
-  "share-invite-instructions": "Follow our invitation instructions on GitHub:",
-  "share-invite-trouble": "Having trouble accessing the invitation link?",
+  "share-invite-intro": "You’re invited to Outline! Follow these instructions to use my server to access the open internet on any device, no matter where you are. No personal information will be collected.",
+  "share-invite-step-one": "1) Download and install the Outline app for your device:",
```
[the remark this answers is not in the document]
Comment (daniellacosse) [reply src/server_manager/messages/en.json:269]: I didn't know if you wanted the `)` - sure

File: src/server_manager/messages/master_messages.json [hunk 5]
```diff
@@ -1243,9 +1243,69 @@
...
+        "content": "{openLink}"
+      }
+    }
+  },
+  "share_invite_step_two": {
+    "message": "2) Copy the following access key. In the Outline app, tap the \"+\" sign on the top bar and paste the access key:",
+    "description": "TODO"
+  },
+  "share_invite_step_three": {
+    "message": "3) Press the \"ADD SERVER\" button to confirm.",
```
[the remark this answers is not in the document]
Comment (daniellacosse) [reply src/server_manager/messages/master_messages.json:1303]: Yeah... if this were a monorepo...

File: src/server_manager/web_app/ui_components/outline-share-dialog.ts [hunk 6]
```diff
@@ -112,24 +112,51 @@ Polymer({
...
         <ol>
```
[the remark this answers is not in the document]
Comment (daniellacosse) [reply src/server_manager/web_app/ui_components/outline-share-dialog.ts:117]: Oh derp, I thought I put a `ul` here

File: src/server_manager/web_app/app.ts (hunk 3 above)
Comment (daniellacosse) [reply src/server_manager/web_app/app.ts:297]: Ah, from our email exchange, I thought we just wanted to quickly point this to that location in the interest of time. Sure. I'll need designs.

Comment (daniellacosse) [reply src/server_manager/web_app/app.ts]: Reverted.

**Review by fortuna — COMMENTED** [review]
Can you share a screenshot of the instructions?

File: src/server_manager/web_app/ui_components/outline-share-dialog.ts [hunk 7]
```diff
@@ -112,23 +112,56 @@ Polymer({
...
           <li>
-            [[localize('share-invite-copy-access-key')]]
-            <a href="{{accessKey}}">{{accessKey}}</a>
+            [[localize('share-invite-install-outline-step')]]
+            <ul>
+              <li>
+                <a href="https://play.google.com/store/apps/details?id=org.outline.android.client"
+                  >[[localize('share-invite-android')]]</a
+                >
+                (<a href="https://s3.amazonaws.com/outline-releases/client/Outline-Client.apk"
```
Comment (fortuna) [inline src/server_manager/web_app/ui_components/outline-share-dialog.ts:124]: Broken link

File: src/server_manager/web_app/ui_components/outline-share-dialog.ts [hunk 8]
```diff
@@ -112,23 +112,56 @@ Polymer({
...
-        <p>[[localize('share-invite-trouble')]]</p>
+        <p>[[localize('share-invite-intro')]]</p>
         <ol>
           <li>
-            [[localize('share-invite-copy-access-key')]]
-            <a href="{{accessKey}}">{{accessKey}}</a>
+            [[localize('share-invite-install-outline-step')]]
+            <ul>
+              <li>
+                <a href="https://play.google.com/store/apps/details?id=org.outline.android.client"
```
Comment (fortuna) [inline src/server_manager/web_app/ui_components/outline-share-dialog.ts:121]: It seems links are not preserved when sharing. We need to spell them out. See the suggestion in your doc.

**Review by fortuna — CHANGES_REQUESTED** [review]

File: src/server_manager/web_app/ui_components/outline-share-dialog.ts (hunk 8 above)
Comment (fortuna) [reply src/server_manager/web_app/ui_components/outline-share-dialog.ts:121]: Formatting may be lost too. This is what I get when I copy the instructions hare:

You’re invited to Outline! Follow these instructions to use my server to access the open internet on any device, no matter where you are. No personal information will be collected.

Download and install the Outline app for your device:
Android: [Play Store](https://play.google.com/store/apps/details?id=org.outline.android.client) - if blocked, try downloading the [APK file](https://s3.amazonaws.com/outline-releases/client/Outline-Client.apk) directly.
iOS: [App Store](https://itunes.apple.com/app/outline-app/id1356177741) 
Windows: [installer .exe](https://s3.amazonaws.com/outline-releases/client/Outline-Client.exe) - once downloaded, double click to launch.
macOS: [App Store](https://itunes.apple.com/app/outline-app/id1356178125) ([Mirror](https://s3.amazonaws.com/outline-releases/client/Outline-Client.dmg)) 
Linux: [AppImage file](https://s3.amazonaws.com/outline-releases/client/Outline-Client.AppImage) - once downloaded, [make it executable](https://docs.appimage.org/introduction/quickstart.html), then double click to launch.
Scan this QR code to have the service added to your Outline app:



If the QR code won’t work, copy the following access key. In the Outline app, tap the "+" sign on the top bar and paste the access key:


ss://Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNTptWk82N0dLODdzNXg@159.203.76.146:443/?outline=1#Canary


Press the "ADD SERVER" button to confirm.
A card will be added to your app. Tap "Connect" to get connected with the new service you added.

Comment (daniellacosse) [reply src/server_manager/web_app/ui_components/outline-share-dialog.ts:121]: Yeah, different clients support different paste "types." You can't paste a hyperlink with an alternative name directly into google docs, for example.

File: src/server_manager/messages/en.json [hunk 9]
```diff
@@ -263,12 +263,22 @@
...
   "share-description": "Copy this invitation and send it from a communication tool you trust. {openLink}Need help?{closeLink}",
-  "share-invite": "You’re invited to connect to my Outline server. Use it to access the open internet, no matter where you are. Follow the instructions on your invitation link below to download the Outline App and get connected.",
   "share-invite-copied": "Copied to clipboard",
   "share-invite-copy": "Copy invitation",
-  "share-invite-copy-access-key": "Copy your access key:",
-  "share-invite-instructions": "Follow our invitation instructions on GitHub:",
-  "share-invite-trouble": "Having trouble accessing the invitation link?",
+  "share-invite-intro": "You’re invited to Outline! Follow these instructions to use my server to access the open internet on any device, no matter where you are. No personal information will be collected.",
+  "share-invite-install-outline-step": "Download and install the Outline app for your device:",
+  "share-invite-android": "Android",
```
[the remark this answers is not in the document]
Comment (daniellacosse) [reply src/server_manager/messages/en.json:270]: I don't know if certain languages would translate these differently, based on the context. Japanese, for example, has the concept of a "topic" that, once established, is inferred until the topic is changed. I don't think we can assume this string will be a simple noun in every language.

File: src/server_manager/web_app/ui_components/outline-share-dialog.ts (hunk 8 above)
Comment (daniellacosse) [reply src/server_manager/web_app/ui_components/outline-share-dialog.ts:121]: How's this?

Comment (daniellacosse) [reply src/server_manager/web_app/ui_components/outline-share-dialog.ts:121]: You’re invited to Outline! Follow these instructions to use my server to access the open internet on any device, no matter where you are. No personal information will be collected.
1) Download and install the Outline app for your device:
  - Android: https://play.google.com/store/apps/details?id=org.outline.android.client
  - iOS: https://itunes.apple.com/app/outline-app/id1356177741 
  - Windows: https://s3.amazonaws.com/outline-releases/client/Outline-Client.exe - once downloaded, double click to launch.
  - macOS: https://itunes.apple.com/app/outline-app/id1356178125
  - Linux: https://s3.amazonaws.com/outline-releases/client/Outline-Client.AppImage - once downloaded, make it executable, then double click to launch.
2) Copy the following access key: ss://[REDACTED]/?outline=1#Canary
3) In the Outline app, tap the "+" sign on the top bar and paste the access key.
4) Press the "ADD SERVER" button to confirm.
5) A card will be added to your app. Tap "Connect" to get connected with the new service you added.

Comment (daniellacosse) [reply src/server_manager/web_app/ui_components/outline-share-dialog.ts:121]: github formats this somewhat: 

You’re invited to Outline! Follow these instructions to use my server to access the open internet on any device, no matter where you are. No personal information will be collected.
1) Download and install the Outline app for your device:
  - Android: https://play.google.com/store/apps/details?id=org.outline.android.client
  - iOS: https://itunes.apple.com/app/outline-app/id1356177741 
  - Windows: https://s3.amazonaws.com/outline-releases/client/Outline-Client.exe - once downloaded, double click to launch.
  - macOS: https://itunes.apple.com/app/outline-app/id1356178125
  - Linux: https://s3.amazonaws.com/outline-releases/client/Outline-Client.AppImage - once downloaded, make it executable, then double click to launch.
2) Copy the following access key: ss://Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNTptWk82N0dLODdzNXg@159.203.76.146:443/?outline=1#Canary
3) In the Outline app, tap the "+" sign on the top bar and paste the access key.
4) Press the "ADD SERVER" button to confirm.
5) A card will be added to your app. Tap "Connect" to get connected with the new service you added.

Comment (daniellacosse) [reply src/server_manager/web_app/ui_components/outline-share-dialog.ts:121]: github formats this slightly :/

You’re invited to Outline! Follow these instructions to use my server to access the open internet on any device, no matter where you are. No personal information will be collected.
1) Download and install the Outline app for your device:
  - Android: https://play.google.com/store/apps/details?id=org.outline.android.client
  - iOS: https://itunes.apple.com/app/outline-app/id1356177741 
  - Windows: https://s3.amazonaws.com/outline-releases/client/Outline-Client.exe - once downloaded, double click to launch.
  - macOS: https://itunes.apple.com/app/outline-app/id1356178125
  - Linux: https://s3.amazonaws.com/outline-releases/client/Outline-Client.AppImage - once downloaded, make it executable, then double click to launch.
2) Copy the following access key: ss://[REDACTED]/?outline=1#Canary
3) In the Outline app, tap the "+" sign on the top bar and paste the access key.
4) Press the "ADD SERVER" button to confirm.
5) A card will be added to your app. Tap "Connect" to get connected with the new service you added.

File: src/server_manager/messages/master_messages.json [hunk 10]
```diff
@@ -1244,8 +1244,13 @@
...
+    "description": "This string appears as an email invitation instructing new Outline users how to set up and connect to Outline.",
+    "placeholders": {
+      "ACCESS_KEY": {
+        "content": "{accessKey}"
```
Comment (fortuna) [inline src/server_manager/messages/master_messages.json:1251]: Can you insert the access key as a link? It may work in some cases, or prevent auto-linking from doing the wrong thing.

File: src/server_manager/messages/master_messages.json [hunk 11]
```diff
@@ -1244,8 +1244,13 @@
...
+    "message": "You're invited to Outline! Follow these instructions to use my server to access the open internet on any device, no matter where you are. No personal information will be collected.&#10;1) Download and install the Outline app for your device:&#10;&#10;  - Android: https://play.google.com/store/apps/details?id=org.outline.android.client&#10;  - iOS: https://itunes.apple.com/app/outline-app/id1356177741&#10;  - Windows: https://s3.amazonaws.com/outline-releases/client/Outline-Client.exe - once downloaded, double click to launch.&#10;  - macOS: https://itunes.apple.com/app/outline-app/id1356178125&#10;  - Linux: https://s3.amazonaws.com/outline-releases/client/Outline-Client.AppImage - once downloaded, make it executable, then double click to launch.&#10;&#10;2) Copy the following access key.&#10;&#10;  $ACCESS_KEY$&#10;&#10;3) In the Outline app, tap the \"+\" sign on the top bar and paste the access key.&#10;&#10;4) Press the \"ADD SERVER\" button to confirm.&#10;&#10;5) A card will be added to your app. Tap \"Connect\" to get connected with the new service you added.",
```
Comment (fortuna) [inline src/server_manager/messages/master_messages.json:1247]: Do translators understand `&#10;`? It looks harder to read and easy to break it accidentally.
Perhaps it's better to use `<br>` for clarity.

File: src/server_manager/messages/master_messages.json [hunk 12]
```diff
@@ -1244,8 +1244,13 @@
     }
   },
   "share_invite": {
```
Comment (fortuna) [inline src/server_manager/messages/master_messages.json:1246]: I noticed that it's hard to tell which messages are text and which are html. I wish we named all HTML messages with a `_html` prefix, like `share_invite_html`.

File: src/server_manager/messages/master_messages.json (hunk 11 above)
Comment (fortuna) [inline src/server_manager/messages/master_messages.json:1247]: What are the guidelines for translating longer text like this? Will formatting be preserved?

Comment (daniellacosse) [reply src/server_manager/messages/master_messages.json]: I don't know. I'll look into it once my laptop has recharged...

Comment (daniellacosse) [reply src/server_manager/messages/master_messages.json]: Okay, seems substituting out the `<br>` for `LINE_BREAK` should mark that text as non-translatable. Updated.

**Review by fortuna — APPROVED** [review]
